tools/ExtensionLibrary/DevHome.ExtensionLibrary/ViewModels/InstalledPackageViewModel.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Copyright (c) Microsoft Corporation and Contributors
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.ObjectModel;
+using System.Threading.Tasks;
+using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Input;
+using DevHome.Common.Contracts;
+using DevHome.Common.Extensions;
+using Microsoft.UI.Xaml;
+using Windows.ApplicationModel;
+using Windows.System;
+
+namespace DevHome.ExtensionLibrary.ViewModels;
+public partial class InstalledExtensionViewModel : ObservableObject
+{
+    [ObservableProperty]
+    private string _displayName;
+
+    [ObservableProperty]
+    private string _packageFullName;
+
+    [ObservableProperty]
+    private bool _hasSettingsProvider;
+
+    [ObservableProperty]
+    private bool _isExtensionEnabled;
+
+    partial void OnIsExtensionEnabledChanging(bool value)
+    {
+        if (IsExtensionEnabled != value)
+        {
+            Task.Run(() =>
+            {
+                var localSettingsService = Application.Current.GetService<ILocalSettingsService>();
+                return localSettingsService.SaveSettingAsync(PackageFullName + "-ExtensionDisabled", !value);
+            }).Wait();
+        }
+    }
+
+    public InstalledExtensionViewModel(string displayName, string packageFullName, bool hasSettingsProvider)

```

Comment:
I believe full name works here. Family name would be better for mapping purposes, but I don't see any problems with using full name that would warrant changing it.

Reply:
Changing to just take ExtensionUniqueId -- we can always add more later if we want and it just uses that one for now.